keep/common/management/commands/update_https.py

```diff
@@ -0,0 +1,214 @@
+    https string in the target_uri.
+
+    The command will also generate a summary report in which you can
+    find the objects that were updated, failed to be updated, as well
+    as any errors (exceptions) that may have occured.
+
+    Dry run is an option that only generates the report without applying
+    any changes. Please be sure that you know what you are doing, and
+    have the Fedora and Pidman environments set up correctly before
+    you proceed. You can use '-n' or '--dry-run' to enable the dry run
+    mode.
+
+    """
+
+    # dry-run option declaration
+    option_list = BaseCommand.option_list + (
+        make_option('--dry-run',
+            '-n',
+            action='store_true',
+            help='Dry run the command to get an output but with no changes applied'),
+    )
+
+    def handle(self, *args, **kwargs):
+        # disable info messages in the console
+        logging.getLogger('requests').setLevel(logging.CRITICAL)
+
+        # initialize an interruption flag
+        self.interrupted = False
+
+        # initialize a Pidman client object
+        self.pidman = self.get_pidman()
+
+        # send greeting message with instructions
+        self.stdout.write(self.greetings)
+
+        # initialize dry-run flag to false
+        self.is_dry_run = False
+
+        # dry run notice
+        if kwargs.get('dry_run', True):
+            dry_run_notice = """
+            [DRY-RUN ACTIVATED] Currently you are running the script
+            in dry-run mode. It means that no changes will be applied
+            to any of the repositories except for that the script will
+            still generate statistics about objects in your repositories.
+
+            """
+            sys.stdout.write(dry_run_notice)
+            self.is_dry_run = True
+
+        # verify environments
+        environment_notice = """
+        Please confirm that you would like to proceed with the
+        following environments:
+        PIDMAN: %s
+        """ % (settings.PIDMAN_DOMAIN)
+        self.stdout.write(environment_notice)
+
+        # confirm to proceed
+        confirmation = raw_input("Verify above information and type 'yes' to proceed: ")
+        if (confirmation != "yes"):
+            self.stdout.write("Answer is not 'yes'. Operation aborted.")
+            sys.exit(1)
+
+        # setup output directory paths
+        self.current_time_string = time.strftime("%Y%m%d-%H%M%S", time.localtime())
+        self.output_path = "%s/%s/%s/" % (os.getcwd(), "tmp", self.current_time_string)
+        self.error_path = "%s/errors/" % self.output_path
+
+        # create output directories
+        if not os.path.exists(self.output_path):
+            os.makedirs(self.output_path)
+        if not os.path.exists(self.error_path):
+            os.makedirs(self.error_path)
+
+        # create log files
+        summary_log_path = open("%s/%s" % (self.output_path, "summary.csv"), 'wb')
```

Comment:
That's a nice suggestion and it could definitely make the CSV handling cleaner but unfortunately it doesn't support unicode and might introduce some unreadable texts. The wrapper that I used supports unicode and also seems like it is the one that we have been using in the past.